src/iminuit/cost.py

```diff
@@ -676,6 +787,77 @@ def __init__(self, n, xe, model, verbose):
         super().__init__(describe(model)[1:], self._prepare_data(n), verbose)
 
 
+class BarlowBeestonLite(MaskedCost):
+    """
+    Binned cost function for a template fit with uncertainties on the template.
+
+    Compared to the Beeston-Barlow method, the lite method uses one nuisance
+    parameter per bin instead of one nuisance parameter per component per bin,
+    and replaces the Poisson constraint with a Gaussian on the multiplicative
+    nuisance parameter.
+
+    J.S. Conway, PHYSTAT 2011, https://doi.org/10.48550/arXiv.1103.0354
+    P. Mandrik, https://arxiv.org/abs/1708.07708
+    """
+
+    __slots__ = ("_templates", "_sum_templates")
+
+    def __init__(
+        self, n, templates, name: _tp.Collection[str] = None, verbose: int = 0
+    ):
+        """
+        Initialize cost function with data and model.
+
+        Parameters
+        ----------
+        n : array-like
+            Histogram counts.
+        templates : array-like
+            TODO
+        name : collection of str, optional
+            Parameter names.
+        verbose : int, optional
+            Verbosity level. 0: is no output (default).
+            1: print current args and negative log-likelihood value.
+        """
+        n = np.atleast_1d(n)
+        templates = np.atleast_2d(templates)
+        M = len(templates)
+        if M < 1:
+            raise ValueError("at least one template is required")
+        if n.shape != templates.shape[1:]:
+            raise ValueError("shapes of n and templates do not match")
+        self._templates = templates
```

Comment:
I would add a scaling of the templates to the data. The way we did it is like follows:

```python
sum_n = np.sum(n)
sum_templates = np.sum(templates, axis = 1)
inv_frac = (sum_n / sum_templates) 
#might need to reshape this array to make sure multiplication is handled correctly
#inv_frac = (sum_n / sum_templates).reshape((len(templates), 1))
self.__variances = inv_frac * inv_frac * np.array(templates)
self.__templates = inv_frac * np.array(templates) 
```
and this would mean you don't need to worry about scaling later on.

Reply:
I will probably follow this suggestion, but let's make sure first that the implementation is correct and produces reasonable results.